### ISSUE TITLE
[a11y]: slider tooltip doesn't round values for decimal steps

### ISSUE THREAD

**Issue by chadluo** [carbon-design-system/carbon#21610]
### Package

carbon-components

### Browser

Chrome

### Operating System

MacOS

### Package version

latest

### Automated testing tool and ruleset

Storybook

### Description

For non integer steps, the value in the tooltip is not rounded, leading to the [floating point issue](https://0.30000000000000004.com/) which looks confusing to the user.

https://web-components.carbondesignsystem.com/?path=/story/components-slider--slider-with-hidden-inputs&args=max:1;step:0.1

<img src="https://github.com/user-attachments/assets/cc3ed5e4-dd25-4b3c-8559-b6aa8c2d550f" />

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/story/components-slider--slider-with-hidden-inputs&args=max:1;step:0.1

### Steps to reproduce

1. create a slider, enabling tooltips
2. use non integer steps, for example `0.1`
3. drag the knob and observe the tooltip

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@chadluo Thanks for reporting. To whoever works on this - we should for sure find a way to cover this in a test. It's very basic functionality that should be stable.

**Comment by filiiipeee** [comment]
Hey, if possible, I would like to work on this bug!
